docs/Handling_Failed_Nodes.md

```diff
@@ -0,0 +1,72 @@
+# Dealing with Failed Nodes
+
+Nodes can be easily removed from a Clearwater deployment by following the instructions for [elastic scaling](Clearwater_Elastic_Scaling.md). When scaling down the remaining nodes are informed that a node is leaving and take appropriate action. However sometimes a node may fail unexpectedly. If this happens and the node cannot be recovered (for example the virtual machine has been deleted) the remaining nodes must be manually informed of the failure. This article explains how to do this.
+
+## Removing a Failed Node
+
+If a node permanently fails scaling the deployment up and down may stop working, or if a scaling operation is in progress it may get stuck (because other nodes in the tier will wait forever for the failed node to react). To recover from this situation the failed node should be removed from the deployment using the following steps:
+
+* Remove the node from the underlying etcd cluster. To do this:
+    * Run `. /etc/clearwater/config` to access clearwater configuration variables.
+    * Run `etcdctl -C $local_ip:4000 cluster-health` and make a note of the ID of the failed node.
+    * Run `etcdctl -C $local_ip:4000 member list` to check that the failed node reported is the one you were expecting (by looking at it's IP address).
+    * Run `etcdctl -C $local_ip:4000 member remove <ID>`, replacing `<ID>` with the ID learned above.
+* Remove the failed node from any back-end data store clusters it was a part of (see Removing a Node From a Data Store).
+
+## Multiple Failed Nodes
+
+If your deployment loses half or more of its nodes permanently, it loses "quorum" which means that the underlying etcd cluster becomes read-only. This means that scaling up and down is not possible and changes to shared config cannot be made. It also means that the steps for removing a single failed node won't work. This section describes how to recover from this state.
+
+In this example, your initial cluster consists of servers A, B, C, D, E and F. D, E and F die permanently, and A, B and C enter a read-only state (because they lack quorum). Recent changes may have been permanently lost at this point (if they were not replicated to the surviving nodes).
+
+You should follow this process completely - the behaviour is unspecified if this process is started but not completed. It is always safe to restart this process from the beginning (for example, if you encounter an error partway through).
+
+To recover from this state:
+
+* stop etcd on A, B and C by running `sudo service clearwater-etcd stop`
+* create a new cluster, only on A, by:
+    * editing `etcd_cluster` in `/etc/clearwater/local_config` to just contain A's IP (e.g. `etcd_cluster=10.0.0.1`)
+    * running `sudo service clearwater-etcd force-new-cluster`. This will warn that this is dangerous and should only be run during this process; choose to proceed.
+    * running `etcdctl -C 10.0.0.1:4000 member list` to check that the cluster only has A in
```

Comment:
If we don't address my comment above, we should be consistent in etcdctl example commands and use $local_ip here instead of 10.0.0.1.